Lib/pdb.py

```diff
@@ -752,7 +752,7 @@ def checkline(self, filename, lineno):
         """
         # this method should be callable before starting debugging, so default
         # to "no globals" if there is no current frame
-        globs = self.curframe.f_globals if hasattr(self, 'curframe') else None
+        globs = self.curframe.f_globals if getattr(self, "curframe", None) else None
```

Comment:
```suggestion
        globs = self.curframe.f_globals if getattr(self, 'curframe', None) else None
```

Reply:
Actually, should we change it such that the self.curframe lookup happens only once?  I know it's only pdb and performance doesn't matter, but still.

Reply:
I was simply thinking of this:
```
frame = getattr(self, 'curframe', None)
globs = frame.f_gloabls if frame else None
```

Reply:
There are two, no?

  2           0 LOAD_GLOBAL              0 (getattr)
              2 LOAD_FAST                0 (self)
              4 LOAD_CONST               1 ('curframe')
              6 LOAD_CONST               0 (None)
              8 CALL_FUNCTION            3   <--- I guess this calls the getattr on 3 args

and then 

             12 LOAD_FAST                0 (self)
             14 LOAD_ATTR                1 (curframe)

Reply:
> There are two, no?

But `getattr` is a builtin, so there's no `LOAD_ATTR` there, right?

Reply:
I don't know how getattr and LOAD_ATTR are implemented, but my point is that if we do the two-line thing then we need only getattr and no LOAD_ATTR at all.

Reply:
is this current?
20 JUMP_IF_NOT_EXC_MATCH    20 (to 40)

I think Mark make some changes recently to get zero-cost exceptions, I don't know what the bytecode looks like now. But this looks familiar (as in, old).

I wouldn't mind try-except, if it contains only the one attr access.

Reply:
Why are you all looking at the bytecode here? If the issue is that `self.curframe` may raise `AttributeError` then the `hasattr()` solution in the original code should be sufficient. If it's possible that `self.curframe` is None, and getting the `f_globals` attribute raises, then the proposed change makes sense, assuming that `self.curframe` is either None or a frame object.

Reply:
I've already pushed the two-liner, so I'm marking this as resolved :)